Docker.DotNet/DockerPipeStream.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Net.Http.Client;
+using Microsoft.Win32.SafeHandles;
+
+namespace Docker.DotNet
+{
+    internal class DockerPipeStream : WriteClosableStream
+    {
+        private PipeStream _stream;
+        private EventWaitHandle _event = new EventWaitHandle(false, EventResetMode.AutoReset);
+
+        public DockerPipeStream(PipeStream stream)
+        {
+            _stream = stream;
+        }
+
+        public override bool CanCloseWrite => true;
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => true;
+
+        public override long Length
+        {
+            get { throw new NotImplementedException(); }
+        }
+
+        public override long Position
+        {
+            get { throw new NotImplementedException(); }
+
+            set { throw new NotImplementedException(); }
+        }
+
+        [DllImport("api-ms-win-core-file-l1-1-0.dll", SetLastError = true)]
+        private static extern int WriteFile(SafeHandle handle, IntPtr buffer, int numBytesToWrite, IntPtr numBytesWritten, ref NativeOverlapped overlapped);
```

Comment:
This pattern works and seems to be used in corefx.

I'm going to hold off on moving this to NativeMethods until we understand how to support this in Linux.